Include/pyport.h

```diff
@@ -114,17 +114,23 @@ typedef intptr_t        Py_intptr_t;
 /* Py_ssize_t is a signed integral type such that sizeof(Py_ssize_t) ==
  * sizeof(size_t).  C99 doesn't define such a thing directly (size_t is an
  * unsigned integral type).  See PEP 353 for details.
+ * PY_SSIZE_T_MAX is the largest positive value of type Py_ssize_t.
  */
 #ifdef HAVE_PY_SSIZE_T
 
 #elif HAVE_SSIZE_T
 typedef ssize_t         Py_ssize_t;
+#   define PY_SSIZE_T_MAX SSIZE_MAX
 #elif SIZEOF_VOID_P == SIZEOF_SIZE_T
 typedef Py_intptr_t     Py_ssize_t;
+#   define PY_SSIZE_T_MAX INTPTR_MAX
 #else
 #   error "Python needs a typedef for Py_ssize_t in pyport.h."
 #endif
 
+/* Smallest negative value of type Py_ssize_t. */
+#define PY_SSIZE_T_MIN (-PY_SSIZE_T_MAX-1)
```

Comment:
How about changing MIN too?

```
# define PY_SSIZE_T_MIN SSIZE_MIN
...
# define PY_SSIZE_T_MIN INTPTR_MIN
```

Reply:
Uh, it is cleaner but needs 4 definition, not 2.

Reply:
There were two reasons:
1. My system did not have a `SSIZE_MIN`, maybe some issue?
2. We would have to make a `#define` in the windows portion as well, which ok TBH, but we can make do with just 1.

I'm ok with both approaches, what do you think?